### PR TITLE
Fix dockerfile and code example

### LINKED ISSUE #3817: Fix bug on install python packages from private sources
### Issue Type

Inaccurate, misleading, or out-of-date information

### Links to Affected Docs

https://www.astronomer.io/docs/astro/cli/private-python-packages?tab=pypi#install-python-packages-from-private-sources

### The Issue or Context

User experienced errors when mounting a secret while trying to install from artifactory. 

### Fix or Proposed Change

Step 1. Replace

```
FROM stage1 AS stage2
# Install Python packages
ARG PIP_EXTRA_INDEX_URL
RUN --mount=type=secret,id=indexurl \
    PIP_EXTRA_INDEX_URL=$(cat indexurl)

COPY requirements.txt .
USER root
RUN pip install --no-cache-dir -q -r requirements.txt
USER astro
```

With this

```
FROM stage1 AS stage2
COPY requirements.txt .
USER root
# Install Python packages
RUN --mount=type=secret,id=indexurl \
    pip install --no-cache-dir -q -r requirements.txt --extra-index-url=$(cat /run/secrets/indexurl)

USER astro
```

2. Replace in Step 3 for building a custom docker image, change
```
astro dev start --build-secrets id=indexurl,src=$HOME/path/to/indexurl.txt .
```
to
```
astro dev start --build-secrets id=indexurl,src=$HOME/src/astro/indexurl.txt
```

### Additional Notes

https://astronomer.slack.com/archives/CL44PA4DB/p1717525504677449

### Required Reviewers

_No response_

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves #3817

### CODE REVIEW

**Review by bearsandrhinos — APPROVED** [review]
Update is accurate to the working code we tested with a customer

File: astro/cli/private-python-packages.md [hunk 1]
```diff
@@ -178,11 +178,11 @@ Ensure that this file is accessible from your Astro project. You will mount this
...
 1. (Optional) Copy and save any existing build steps in your `Dockerfile`.
 
-2. In your `Dockerfile`, add `AS stage` to the `FROM` line which specifies your Runtime image. For example, if you use Runtime 5.0.0, your `FROM` line would be:
+2. In your `Dockerfile`, add `AS stage` to the `FROM` line which specifies your Runtime image. For example, if you use Runtime X.Y.Z, your `FROM` line is:
 
-   ```text
-   quay.io/astronomer/astro-runtime:5.0.0-base AS stage1
-   ```
+    ```text
+    quay.io/astronomer/astro-runtime:X.Y.Z-base AS stage1
```
Comment (ReadytoRocc) [inline astro/cli/private-python-packages.md:184]: @lzdanski - stages are should not be required when using secrets. We should remove these.

**Review by bearsandrhinos — COMMENTED** [review]
Removed stage and looks good

**Review by bearsandrhinos — APPROVED** [review]

**Comment by lzdanski** [comment]
Received technical review and no substantial content changes, just code example changes. Merging.

**Pull request merged**